nautobot/core/tests/integration/test_view_authentication.py

```diff
@@ -0,0 +1,67 @@
+from django.test import tag
+
+from nautobot.core.testing import TestCase
+from nautobot.core.utils.lookup import get_url_for_url_pattern, get_url_patterns
+
+
+@tag("integration")
+class AuthenticationEnforcedTestCase(TestCase):
+    r"""
+    Test that all\* registered views require authentication to access.
+
+    \* with a very small number of known exceptions such as login and logout views.
+    """
+
+    def test_all_views_require_authentication(self):
```

Comment:
This would be a good test to expose to apps. Not blocking but should we write a generic test for this?

Reply:
I do plan/want to add a generic test to the cookiecutter that takes in the app's urlconf and iterates/validates its url patterns, yeah. I like the idea of implementing the bulk of it in core rather than in the cookiecutter but that'll take some thinking about how to do it.

Reply:
Is there a follow-up issue for that work?

Reply:
https://github.com/nautobot/cookiecutter-nautobot-app/issues/102